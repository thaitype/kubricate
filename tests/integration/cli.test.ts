import { describe, expect, it } from 'vitest';

import { executeKubricate } from '../helpers/execute-kubricate';

const scenarios = [
  { name: 'Run no command', args: [], expectExitCode: 0 },
  { name: 'Run no command with --help option', args: ['--help'], expectExitCode: 0 },
  { name: 'Run no command with --h option', args: ['-h'], expectExitCode: 0 },
];

describe.each(scenarios)('CLI output display ($name)', ({ args, expectExitCode }) => {
  it('should display expected output', async () => {
    const { all, exitCode } = await executeKubricate(args, { all: true, reject: false });

    expect(exitCode).toBe(expectExitCode);
    expect(all).toMatchSnapshot();
  });
});
