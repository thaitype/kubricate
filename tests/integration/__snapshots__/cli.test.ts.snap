// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CLI output display ('Run generate command with unknown optâ€¦') > should display expected output 1`] = `
"[31m[39m
[31mâœ– Error: Unknown option "unknown".[39m
[31m[39m
[36m[34mâ„¹[36m Check the help for a list of available options.[39m
[36m[39m
kbr generate

Generate a stack into yaml files

Options:
      --root     Root directory                                         [string]
      --config   Config file path                                       [string]
      --verbose  Enable verbose output                                 [boolean]
      --silent   Suppress all output                                   [boolean]
      --dry-run  Dry run mode (Not Stable Yet)                         [boolean]
      --outDir   Output directory                   [string] [default: "output"]
      --stdout   Output to stdout                     [boolean] [default: false]
      --filter   Filter stacks or resources by ID (e.g., myStack or myStack.reso
                 urce)                                                   [array]
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]


[31m[31mâœ–[31m Error: Unknown option "unknown"[39m"
`;

exports[`CLI output display ('Run no command with --h option') > should display expected output 1`] = `
"kbr <command>

Commands:
  kbr generate          Generate a stack into yaml files
  kbr secret [command]  Manage secrets with SecretManager

Options:
      --root     Root directory                                         [string]
      --config   Config file path                                       [string]
      --verbose  Enable verbose output                                 [boolean]
      --silent   Suppress all output                                   [boolean]
      --dry-run  Dry run mode (Not Stable Yet)                         [boolean]
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]"
`;

exports[`CLI output display ('Run no command with --help option') > should display expected output 1`] = `
"kbr <command>

Commands:
  kbr generate          Generate a stack into yaml files
  kbr secret [command]  Manage secrets with SecretManager

Options:
      --root     Root directory                                         [string]
      --config   Config file path                                       [string]
      --verbose  Enable verbose output                                 [boolean]
      --silent   Suppress all output                                   [boolean]
      --dry-run  Dry run mode (Not Stable Yet)                         [boolean]
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]"
`;

exports[`CLI output display ('Run no command') > should display expected output 1`] = `
"[31m[39m
[31mâœ– Error: Require <command>.[39m
[31m[39m
[36m[34mâ„¹[36m Check the help for a list of available options.[39m
[36m[39m
kbr <command>

Commands:
  kbr generate          Generate a stack into yaml files
  kbr secret [command]  Manage secrets with SecretManager

Options:
      --root     Root directory                                         [string]
      --config   Config file path                                       [string]
      --verbose  Enable verbose output                                 [boolean]
      --silent   Suppress all output                                   [boolean]
      --dry-run  Dry run mode (Not Stable Yet)                         [boolean]
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]


[31m[31mâœ–[31m Error: Require <command>[39m"
`;
